### PR TITLE
refactor: prepare access conditions for EDT ^0.18

### PR DESCRIPTION
**Pull request by dresslerdemos**
In EDT ^0.18 the following changes took place:

1. The `getAccessCondition` method was renamed to
 `getAccessConditions` and must now return an array of conditions
  (of which all must match for access to be granted). This
  simplifies the further handling of conditions in EDT.
2. Condition paths used in `getAccessConditions` MUST NOT use
  aliases. Aliases allow to expose properties via the JSON:API
  that may have different names or reside in different entities
  than the backing entity. However, this is not needed for the
  access conditions and unnecessarily complicated path handling
  in EDT in the past.
3. As retrieval of resources is now done in the resource type
  itself, `getAccessConditions` does not need to be set to `public`
  visibility anymore, reducing the surface area and complexity of
  resource types. The only exception is the `StatementCluster`
  resource type, which was kept `public` for now.

The dplan core resource types were adjusted accordingly to prepare
for the actual EDT ^0.18 update, while keeping compatibility to
the current version. The necessary changes in the (upcoming) actual
dependency update are thus further reduced.

### How to review/test

I suggest to do a code review via IDE (focusing on the points above) and test inside a project.
I only briefly tested EWM locally (procedure-/statement-/segment-list). If more extensive tests uncover any bugs they are easy to fix.

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/ResourceTypes/StatementFragmentResourceType.php [hunk 1]
```diff
@@ -159,18 +159,18 @@ public function isReferencable(): bool
...
-            return $this->conditionFactory->false();
+            return [$this->conditionFactory->false()];
         }
 
-        return $this->conditionFactory->allConditionsApply(
+        return [
             $this->conditionFactory->propertyHasValue($procedure->getId(), $this->procedure->id),
             $this->conditionFactory->propertyHasValue($procedure->getId(), $this->statement->procedure->id),
-            $this->conditionFactory->propertyHasValue(false, $this->deleted),
+            $this->conditionFactory->propertyHasValue(false, $this->deleted), // FIXME: cdr: this should not work, there is no `deleted` property in `StatementFragment` nor is `deleted` set up as alias in this resource type
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/ResourceTypes/StatementFragmentResourceType.php:172]: Please listen to the fix me and fix it :)

**Review by breierDemos — APPROVED** [review]
lgtm

**Review by AlexMenschoid — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/ResourceTypes/UserResourceType.php [hunk 2]
```diff
@@ -158,10 +158,10 @@ public function getAccessCondition(): PathsBasedInterface
         if ($userAuthorized) {
             // allow access to all users when working inside a procedure to request assignees of statements or segments
             // TODO: split into AssigneeResourceType to differentiate between claiming and normal (more restricted) user accesses
-            return $this->conditionFactory->true();
+            return [];
         }
 
-        return $this->conditionFactory->propertyHasValue($user->getId(), $this->id);
+        return [$this->conditionFactory->propertyHasValue($user->getId(), $this->id)];
```
[the remark this answers is not in the document]
Comment (dresslerdemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/UserResourceType.php:164]: `id` is defined in `DplanResourceType`, which is a parent class of `UserResourceType`.

File: demosplan/DemosPlanCoreBundle/ResourceTypes/StatementFragmentResourceType.php (hunk 1 above)
Comment (dresslerdemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/StatementFragmentResourceType.php]: done in a078b494f949431f495041dbc9ef6c9c8f667c6e

**Review by AlexMenschoid — APPROVED** [review]
lgtm, code review and a bit of quick testing in a project.

**Pull request merged**